editor/plugins/script_editor_plugin.cpp

```diff
@@ -2439,6 +2439,10 @@ void ScriptEditor::save_current_script() {
 }
 
 void ScriptEditor::save_all_scripts() {
+	if (close_without_saving) {
+		return; // TODO: Find a better way of preventing scripts from saving on quit.
+	}
```

Comment:
Why is this needed? `save_all_scripts()` isn't normally called when exiting.

Reply:
This confused me as well, but I did a lot of tests and scripts always seemed to be saved... It's also in the engine logic; there's a loop that goes through all plugins (including script editor) and saves their external data.
If you could help me work out why, that would be greatly appreciated as I really don't like my solution (;

Reply:
I investigated it and it doesn't make any sense xd
This method is called from EditorNode using `save_external_data()`. It's called when you quit the editor, but it doesn't actually save anything. I debugged it, it goes correctly over all lines and even saves scenes when saving builtin-scripts. But when you reopen the project, no script nor any scene is actually saved.
wtf

This is only on vanilla build. With your changes, when you close with Don't Save, this method actually has some effect.

Reply:
Wow, freaky... I did have a go at debugging it but didn't realise it was working that weirdly lol
I'm pretty new to all of this, but I'm guessing one of us should look into it and possibly open a pull request (or modify this one) if we can work out what needs fixing, if anything?